homeassistant/components/rainmachine/__init__.py

```diff
@@ -323,6 +324,28 @@ async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     return unload_ok
 
 
+async def async_migrate_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Migrate old entry."""
+    version = entry.version
+
+    LOGGER.debug("Migrating from version %s", version)
+
+    # 1 -> 2: Unique ID format changed, so delete and re-import:
```

Comment:
We should migrate the existing entries to the new format instead.  The current implementation resulted in all new entity ids.

<img width="301" alt="Screen Shot 2021-11-20 at 09 38 05" src="https://user-images.githubusercontent.com/663432/142732291-029d8cd3-9455-4b75-8fb9-af91acf219c9.png">